shellwhat/checks/check_funcs.py

```diff
@@ -55,7 +57,7 @@ def has_code(state, text, incorrect_msg="The checker expected to find `{{text}}`
 
     if not res:
         _msg = state.build_message(incorrect_msg, fmt_kwargs={ 'text': text })
-        state.do_test(_msg)
+        state.report(Feedback(_msg))
```

Comment:
if whenever state.report is called, `_msg` is always wrapped as `Feedback(_msg)`, then may cleaner to include `Feedback` inside `state.report`?

(may be other reasons not to do it?)

Reply:
looking at protowhat, it looks like Feedback's main job right now outside `state.report`, is to wrap a string being passed to `state.report`.

![image](https://user-images.githubusercontent.com/2574498/57487435-8d163780-727e-11e9-8f84-2e29dfc0221d.png)

Reply:
Some SCT libraries had `do_test(msg: str)`, some `do_test(test: Test[Feedback])`.
The second parameter to `Feedback` wasn't `State` and by constructing `Feedback` as argument there was less passing around of arguments I think.
Looking at the code now, it looks like `state.report` could construct `Feedback(msg, self)`, resulting in cleaner code.